### DISCUSSION TITLE
how to change appstoreconnect app name

### DISCUSSION THREAD

**Discussion by csc-EricWu** [fastlane/fastlane#18531] [General] [answered]
username = "dd@out.com"
    app_identifier = "com.ios.dd"
    ENV["FASTLANE_ITC_TEAM_ID"] = "232323"
    Spaceship::Tunes.login(username)
    Spaceship::Tunes.select_team
    UI.message("Login successful")
    spaceship_app ||= Spaceship::ConnectAPI::App.find(app_identifier)
    UI.user_error!("Couldn't find app '#{app_identifier}' on the account of '#{username}' on iTunes Connect") unless spaceship_app
    UI.message("#{spaceship_app.name} #{app_identifier} app_id : #{spaceship_app.id}")
    #AppInfoLocalization
    localInfo = spaceship_app.update(attributes: { name: "sdfsd3434" })

I got the mistake

An attribute in the provided entity is not allowed for this request - The attribute 'name' can not be included in a 'UPDATE' request - /data/attributes/name

**Comment by Elmirakuphal** [comment, marked as answer]
The only way you can edit a App Name (assuming the app has been approved) is by creating a new version, at that point you can modify it. Apple has removed the tables that once referenced what could be updated during what statuses. Now Apple has outlined exactly what CAN be change while the app is in the 'Ready for Sale' status. Anything else will require a new version to be created.

[PrepaidGiftBalance](https://www.prepaidgiftbalance.net/)
